### PR TITLE
Migration MacOS CI to Github Action python-only part

### LINKED ISSUE #2656: Migrate Mac CI: Migrate Remaining Python test steps

### PR DESCRIPTION
**Pull request by FirelightFlagboy**
closes #2656

### CODE REVIEW

File: .github/workflows/macos-ci.yml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        run: poetry install -E core -E backend
+
+      - name: Generate PyQT GUI data
+        run: poetry run python misc/generate_pyqt.py
+
+      - name: Test python
+        run: >
+          poetry run pytest
+          -n auto
+          --log-level=DEBUG
```
Comment (touilleMan) [inline .github/workflows/macos-ci.yml:46]: better to put that in a `pytest-args` env config to stay consistent with windows/linux jobs

**Review by touilleMan — COMMENTED** [review]
What about the messages explaining why we don't do much in the macOS tests ?
 
https://github.com/Scille/parsec-cloud/blob/f040851f7c63f3cff01d3f96aad4366f953d1725/.azure-pipelines.yml#L430-L437

File: .github/workflows/macos-ci.yml (hunk 1 above)
Comment (FirelightFlagboy) [reply .github/workflows/macos-ci.yml:46]: not needed for the moment, but will do latter

**Comment by FirelightFlagboy** [comment]
> What about the messages explaining why we don't do much in the macOS tests ?

As you see the tests are working with `-n auto` and we're not on `azure-pipeline` so for me the comment is out-of-date

**Review by touilleMan — APPROVED** [review]

**Pull request merged**